torch/package/package_exporter.py

```diff
@@ -852,15 +853,89 @@ def _execute_dependency_graph(self):
+
+        def print_pretty(
+            category: str,
+            entries: List[str],
+            indentation=0,
+            recurse=True,
+            bullet_char="-",
+        ):
+            tab = "    "
+            tabs = tab * indentation
+            print(f"{tabs}{bullet_char} {category}:")
+            tabs += tab
+            for item in sorted(entries):
+                print(f"{tabs}{item}")
+                if recurse and self.verbose > VerbosityLevel.PACKAGE_INFO:
+                    successors = self.dependency_graph._succ[item].keys()
+                    if successors:
+                        print_pretty(
+                            "relies on",
+                            successors,
+                            indentation=2,
+                            recurse=False,
+                            bullet_char="*",
+                        )
+                    predecessors = self.dependency_graph._pred[item].keys()
+                    if predecessors:
+                        print_pretty(
+                            "included in",
+                            predecessors,
+                            indentation=2,
+                            recurse=False,
+                            bullet_char="*",
+                        )
+
+        interned_mods = []
+        explicitly_externed_mods = []
+        implicitly_externed_mods = []
+        mocked_mods = []
+        pickled_objs = []
+        invalid_nodes = []
+        denied_nodes = []
+
+        for module_name, attrs in self.dependency_graph.nodes.items():
+            action = attrs.get("action")
+
+            if action == _ModuleProviderAction.EXTERN:
+                if attrs.get("implicityly_externed"):
+                    implicitly_externed_mods.append(module_name)
+                else:
+                    explicitly_externed_mods.append(module_name)
+            elif action == _ModuleProviderAction.MOCK:
+                mocked_mods.append(module_name)
+            elif action == _ModuleProviderAction.INTERN:
+                if attrs.get("is_pickle"):
+                    pickled_objs.append(module_name)
+                else:
+                    interned_mods.append(module_name)
+            elif action == _ModuleProviderAction.DENY:
+                denied_nodes.append(module_name)
+            else:
+                invalid_nodes.append(module_name)
+
+        print("Package Contents:")
+        print_pretty("interned modules", interned_mods)
+        print_pretty("explicitly externed modules", explicitly_externed_mods)
+        print_pretty("implicitly externed modules", implicitly_externed_mods)
+        print_pretty("mocked modules", mocked_mods)
+        print_pretty("pickles", pickled_objs)
+        print_pretty("denied modules which are depended upon", denied_nodes)
+        print_pretty("invalid objects", invalid_nodes)
+        if self.verbose > VerbosityLevel.PACKAGE_INFO:
```

Comment:
There's some duplication in intent between this code and the error messaging code. Both will print listings of modules grouped by their characteristics (broken/denied/etc.). We should decide what information we want to go into the verbose logs vs. the error message.